src/connected_component.rs

```diff
@@ -0,0 +1,215 @@
+use std::collections::{ HashSet };
+use weighted_graph::{ Graph, NodeId };
+use dijkstra::shortest_path;
+
+pub fn reduce_to_largest_connected_component(graph: Graph) -> Graph {
+    let untested_nodes = node_ids(&graph);
+    reducer(graph, untested_nodes, vec![])
+}
+
+fn reducer(graph: Graph, untested_nodes: HashSet<NodeId>, mut results: Vec<HashSet<NodeId>>) -> Graph {
```

Comment:
None of those unwraps should produce errors, but this can be better expressed using `match`.